autoware_adapi_v1_msgs/vehicle/msg/DoorGroup.msg

```diff
@@ -0,0 +1,5 @@
+uint8 ALL = 1
+uint8 GET_ON = 2
+uint8 GET_OFF = 3
+
+uint8 group
```

Comment:
What are these for?

Is this to show the reason why the door group was opened?

Or is this for buses where you should enter from front door and exit from back doors?

Reply:
This is for a use case like getting on and off a bus. The application can control the doors regardless of the layout. I think it will be useful for the door opening and closing API that will be created later.

Reply:
I still don't understand, GET_ON and GET_OFF means doors should be opened from the passenger's point of view :thinking: And from these doors, some passengers could get on to the bus, some could get off from the bus.

Reply:
For example, one vehicle has one door on each side, and another vehicle has two doors on each side. The application does not know the position and number of doors to control when getting on/off.

In most cases, the application can determine left and right from the position of the doors and open all the doors on either side, but maybe one of them is the driver's seat and not for the passenger. Since this information depends on the vehicle, API provide it as a group of doors.

I think there is also the idea that it is not included until the use case is not clear. What do you think?

Reply:
I had a meeting with @xmfcx and decided to delete `ALL` constant as a result. I was thinking of using this group for door operations, but I'm going to standardize the interface with a door index. For example, an application creates the following arrays to open doors given the following information in the door layout.

```
DoorLayout[0]: [GET_ON, GET_OFF]
DoorLayout[1]: []
DoorLayout[2]: [GET_ON]
DoorLayout[3]: [GET_OFF]

GET_ON:  [true, false, true, false]
GET_OFF: [true, false, false, true]
```